api/services/media_source.py

```
@@ -0,0 +1,375 @@
+        if not os.access(self.file_path, os.R_OK):
+            return False, f"❌ Bestand niet leesbaar: {os.path.basename(self.file_path)}"
+        
+        # Check if it's a video file
+        valid_extensions = {'.mp4', '.avi', '.mov', '.mkv', '.webm', '.flv', '.wmv', '.m4v'}
+        ext = os.path.splitext(self.file_path)[1].lower()
+        
+        if ext not in valid_extensions:
+            if ext:
+                return False, f"❌ Bestandstype '{ext}' wordt niet ondersteund. Gebruik: MP4, AVI, MOV, MKV, WebM"
+            else:
+                return False, f"❌ Geen geldig bestandstype gevonden. Upload een video bestand (MP4, AVI, MOV, etc.)"
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)